packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -323,9 +324,30 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
   };
 
   generateResolvers = (context: TransformerContextProvider): void => {
+    const dataSourceMapping: Record<string, string> = {};
     this.resourceGeneratorMap.forEach((generator) => {
       generator.generateResources(context);
+      const ddbDatasources = Object.entries(generator.getDatasourceMap()).filter(
+        ([, datasource]) => datasource.ds.type === 'AMAZON_DYNAMODB',
+      );
+      ddbDatasources.forEach(([modelName, datasource]) => {
+        if (datasource.ds.dynamoDbConfig && !cdk.isResolvableObject(datasource.ds.dynamoDbConfig)) {
+          dataSourceMapping[modelName] = datasource.ds.dynamoDbConfig.tableName;
+        } else {
+          console.warn(
+            `Could not resolve table name for ${modelName}. DataSourceMappingOutput is incomplete. Please manually add ${modelName} to the mapping for your migration.`,
+          );
+        }
+      });
     });
+    if (context.transformParameters.enableGen2Migration && context.transformParameters.enableTransformerCfnOutputs) {
+      const { scope } = context.stackManager;
+      new cdk.CfnOutput(cdk.Stack.of(scope), 'DataSourceMappingOutput', {
+        value: cdk.Stack.of(scope).toJsonString(dataSourceMapping),
+        description: 'Mapping of model name to data source table name.',
```

Comment:
nit: Is it possible to add the environment name to this description? (I'm not sure where we get that from, so it's a nice to have, not a must have)